packages/test/test-gc-sweep-tests/src/channelHandler.ts

```diff
@@ -0,0 +1,234 @@
+    ) {
+		super(channel);
+		this.actions.push(async () => {
+			this.createRandomStroke();
+		});
+		this.actions.push(async () => {
+			let strokes = channel.getStrokes();
+			if (strokes.length === 0) {
+				this.createRandomStroke();
+				strokes = channel.getStrokes();
+			}
+			const stroke = this.random.pick(strokes);
+			const point: IInkPoint = {
+				x: this.random.integer(1, 1000),
+				y: this.random.integer(1, 1000),
+				time: this.random.integer(1, 1000),
+				pressure: this.random.integer(1, 1000),
+			};
+
+			channel.appendPointToStroke(point, stroke.id);
+		});
+		this.actions.push(async () => channel.clear());
+	}
+
+	private createRandomStroke() {
+		const color: IColor = {
+			r: this.random.integer(0, 256),
+			g: this.random.integer(0, 256),
+			b: this.random.integer(0, 256),
+			a: this.random.integer(0, 100),
+		};
+		const pen: IPen = {
+			color,
+			thickness: this.random.integer(1, 10),
+		};
+		this.channel.createStroke(pen);
+	}
+
+	public async executeRandomNonHandleOp(): Promise<void> {
+		assert(this.actions.length === 0, "this.actions should have actions added!");
+		const action = this.random.pick(this.actions);
+		await action();
+	}
+}
+
+export class SharedCellHandler extends HandleOpHandler {
+	constructor(
+        protected readonly channel: SharedCell<IFluidHandle>,
+    ) {
+		super(channel);
+		this.removeHandleActions.push(async () => this.removeHandle());
+	}
+
+	protected async addHandle(handle: IFluidHandle): Promise<IFluidHandle | undefined> {
+		const potentialHandle = this.channel.get();
+		this.channel.set(handle);
+		this.handles.set(handle.absolutePath, handle);
+		if (potentialHandle !== undefined) {
+			this.handles.delete(potentialHandle.absolutePath);
+		}
+		return potentialHandle;
+	}
+
+	private removeHandle(): IFluidHandle | undefined {
+		const handle = this.channel.get();
+		this.channel.delete();
+		if (handle !== undefined) {
+			this.handles.delete(handle.absolutePath);
+		}
+		return handle;
+	}
+}
+
+export class SharedCounterHandler extends OpHandler implements IRandomOpHandler {
+    constructor(
+        protected readonly channel: SharedCounter,
+    ) {
+		super(channel);
+	}
+
+	public async executeRandomNonHandleOp(): Promise<void> {
+		this.channel.increment(this.random.integer(1, 10));
+	}
+}
+
+export class SharedMapHandler extends HandleOpHandler implements IRandomOpHandler {
+	constructor(
+        protected readonly channel: SharedMap,
+    ) {
+		super(channel);
+		this.removeHandleActions.push(async () => this.removeHandle());
+	}
+
+	public async executeRandomNonHandleOp(): Promise<void> {
+		this.channel.set("NotAHandleOp", this.random.string(10));
+	}
+
+	protected async addHandle(handle: IFluidHandle): Promise<IFluidHandle | undefined> {
+		// Only a handle should be added where we are using absolute paths.
+		const oldHandle = this.channel.get(handle.absolutePath) as IFluidHandle;
+		this.channel.set(handle.absolutePath, handle);
```

Comment:
Very valid point. I think I'm going to need more logic about where each handle is stored and how it is stored, whether or not it can be removed, and how it can be removed. I haven't gotten there yet.